File: selenium-tests/src/main/java/com/automationrhapsody/observability/pages/HomePage.java
package com.automationrhapsody.observability.pages;

import com.automationrhapsody.observability.TracingWebDriver;
import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;

public class HomePage {

    private TracingWebDriver webDriver;

    public HomePage(TracingWebDriver webDriver) {
        this.webDriver = webDriver;
    }

    private WebElement getFetchPersonsButton() {
        return webDriver.findElement(By.id("test-fetch-persons-button"));
    }

    public CreateNewPersonPage clickCreateNewPersonButton() {
        webDriver.findElement(By.id("test-create-person-button")).click();
        return new CreateNewPersonPage(webDriver);
    }

    public void clickFetchPersonsButton() {
        getFetchPersonsButton().click();
    }

    public String getPersonsCountText() {
        return webDriver.findElement(By.id("test-persons-count-text")).getText();
    }
}

File: cypress-tests/cypress/pages/wrapper/home_page.js
import CreatePersonPage from './create_person_page'

export default class HomePage {
  constructor(tracingCy) {
    this.tracingCy = tracingCy
    this.elements = {
      fetchPersonsButton: () => this.tracingCy.get('#test-fetch-persons-button'),
      createPersonButton: () => this.tracingCy.get('#test-create-person-button')
    }
  }

  goTo() {
    this.tracingCy.visit('http://localhost:3000')
  }

  clickFetchPersonsButton() {
    this.tracingCy.click(this.elements.fetchPersonsButton())
  }

  clickCreatePersonButton() {
    this.tracingCy.click(this.elements.createPersonButton())
    return new CreatePersonPage(this.tracingCy)
  }
}
